src/Helper/BaseHelper.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\IntlBundle\Helper;
+
+use Sonata\IntlBundle\Locale\LocaleDetectorInterface;
+
+/**
+ * BaseHelper provides charset conversion.
+ *
+ * The php Intl extension will always output UTF-8 encoded strings [1]. If the
+ * framework is configured to use another encoding than UTF-8 this will lead to
+ * garbled text. The BaseHelper provides functionality to the other helpers to
+ * convert from UTF-8 to the kernel charset.
+ *
+ * [1] http://www.php.net/manual/en/intl.examples.basic.php
+ *
+ * @author Alexander <iam.asm89@gmail.com>
+ */
+abstract class BaseHelper
```

Comment:
Is this base class needed or it could be a dep instead of extends?

An abstract class only makes sense when there are abstract methods.

Reply:
This is a copy of the Templating/Helper which was extending the helper from symfony-templating.
But indeed, since we're introducing new classes there is a better conception to have here. (If this helper is still useful)

Reply:
the base helper is for the charset functions
i don't know if doing it as dep would be better.

also that would change the constructor parameters maybe wouldn't make them interchangeable?

Reply:
what really makes a class interchangeable with another is a common interface, thats why I asked for that. This would have been easier if the previous class already implemented interfaces.

Reply:
yes, but that would have been a BC

Reply:
Not for the new class.

Reply:
For the new class you can do everything you want.
So the helper can be a service injected in the construct of the new helpers classes

Reply:
hey for the new helpers, i'm allowed to use `Symfony\Contracts\Translation\LocaleAwareInterface` instead of `LocaleDetectorInterface` ? or would that be a BC?

Reply:
You will need to add that dependency on composer.json. 

Is it really worth it the change? What are the benefits?

Reply:
it simplifies our code, and let other modules take care about it better

Reply:
i need to probably deprecate more stuff that isn't needed anymore